### PR TITLE
DevHub desktop example

### PR DESCRIPTION
**Pull request by ravinderjangra**
Fixes: #4, #10, #8 

**DebHub desktop example app .NET Framework. (Console App)** 

**Purpose**:  Developers can use this app as a reference while using the DevHub C# guide.

**Features**:
- Mock authentication process (Using random account credentials).
- Test network authentication flow with the SAFE browser desktop app (using SAFE account).
- Using System registry for URL registration.
- Single instance app. Using IPC Pipes for inter-instance communication.
- Create and retrieve a RandomPrivateMutableData on the network.
- Add/Retrieve entries to created mutable data.
- Update/Delete entries from existing mutable.

### CODE REVIEW

File: SafeDesktopExample/SharedDemoCode/Network/DataOperations.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                var value = await _session.MData.GetValueAsync(_mdinfo, keyToUpdate.Val);
+                await _session.MDataEntryActions.UpdateAsync(entriesHandle, keyToUpdate.Val, newValue.ToUtfBytes(), value.Item2 + 1);
+                await _session.MData.MutateEntriesAsync(_mdinfo, entriesHandle);
+            }
+        }
+
+        private static async Task DeleteEntry()
+        {
+            var keys = await _session.MData.ListKeysAsync(_mdinfo);
+            var keyToDelete = keys[new Random().Next(5)];
```
Comment (krishnaIndia) [inline SafeDesktopExample/SharedDemoCode/Network/DataOperations.cs]: Can we update to use keys.length instead of hard coding as 5?

**Comment by S-Coyle** [comment]
@ravinderjangra Should the README be updated as part of this?

**Comment by ravinderjangra** [comment]
Yes @S-Coyle I will update README as well

**Comment by ravinderjangra** [comment]
changes to WIP as updating to add #8 , #10

**Comment by ravinderjangra** [comment]
The new version of MaidSafe.SafeApp package will be required.

File: SafeDesktopExample/MockNetworkExample/Program.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+        private static async Task Main()
+        {
+            string[] args = Environment.GetCommandLineArgs();
+            Console.WriteLine("SafeNetwork Console Application");
+            try
+            {
+                if (IsApplicationFirstInstance())
+                {
+                    Console.Write("Press Y to use mock safe-browser for authentication : ");
```
Comment (ashwinkumar6) [inline SafeDesktopExample/MockNetworkExample/Program.cs]: lets also mention that anything other than y/Y will use mock network without browser

File: SafeDesktopExample/MockNetworkExample/Program.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                    var key = Console.ReadKey().KeyChar;
```
Comment (ashwinkumar6) [inline SafeDesktopExample/MockNetworkExample/Program.cs]: We could use ReadLine() instead of ReadKey(), to keep it uniform

**Pull request merged**